nodejs/README.adoc

```diff
@@ -0,0 +1,341 @@
++
+.Running the consumer component
+[source]
+----
+$ node consumer.js
+----
++
+You should see the Node.js application start to run and connect to the Kafka instance. However, because you haven't run the producer component yet, the consumer has no country names to display.
++
+If the application fails to run, review the error log in the command-line window and address any problems. Also, review the steps in this quick start to ensure that the application and Kafka topic are configured correctly.
+
+. On your computer, open a second command-line window or tab. On the command line, navigate to the `reactive-example` directory corresponding to the repository that you cloned.
++
+.Navigating to the reactive-example directory
+[source]
+----
+$ cd ~/reactive-example
+----
+
+. Check out the `kafkajs` branch of the repository.
++
+.Checking out the kafkajs branch
+[source]
+----
+$ git checkout kafkajs
+----
+
+. Run the shell script that sets environment variables for the bootstrap server endpoint and client credentials.
++
+.Running the shell script
+[source]
+----
+$ source set-environment.sh
+----
+
+. Navigate to the directory for the producer component. Use Node Package Manager (npm) to install the package for this component.
++
+.Installing the producer component
+[source]
+----
+$ cd producer-backend
+$ npm install
+----
+
+. Run the producer component.
++
+.Running the producer component
+[source]
+----
+$ node producer.js
+----
++
+You should see output like that shown in the following example:
++
+.Example output from the producer component
+[source]
+----
+$ node producer.js
+Ghana
+Réunion
+Guatemala
+Luxembourg
+Mayotte
+Syria
+United Kingdom
+Bolivia
+Haiti
+----
++
+As shown in the example, the producer component starts to run and generate messages that represent country names. The producer writes all messages to the single partition that you specified when you created your Kafka topic.
+
+. Switch back to the first command-line window that you opened.
++
+You should now see that the consumer component displays the same country names generated by the producer, and in the same order, as shown in the following example:
++
+.Example output from the consumer component
+[source]
+----
+$ node consumer.js
+Ghana
+Réunion
+Guatemala
+Luxembourg
+Mayotte
+Syria
+United Kingdom
+Bolivia
+Haiti
+----
++
+The output from both components shows that the Node.js application successfully connected to the Kafka instance. The application is using the Kafka topic you created to produce and consume messages.
+
+. To verify __how__ the Node.js application connected to the Kafka instance, perform the following actions:
+.. In your IDE, open the `producer.js` file.
++
+Observe that the producer component is using the environment variables that you set in the `set-environment.sh` shell script. The values of these environment variables are the bootstrap server endpoint and client credentials that the producer component used to connect to the Kafka instance.
```

Comment:
```suggestion
Observe that the producer component is using the environment variables that you set in the `.env` file. The values of these environment variables are the bootstrap server endpoint and client credentials that the producer component used to connect to the Kafka instance.

```

Reply:
Thanks a lot for making these updates, @mhdawson. I made some other updates on my branch in the meantime, so I'll incorporate your updates also. I should be able to push all those changes sometime Monday and will ask you for a final review then.